### PR TITLE
Add contributing doc

### LINKED ISSUE #279: Consider add CONTRIBUTING.md
... to guide community how to contribute.

### PR DESCRIPTION
**Pull request by jdneo**
resolve #279

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+## First Time Setup
```
Comment (yaohaizh) [inline CONTRIBUTING.md]: First Time Setup ==> Setup

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+1. Fork and clone the repository: `git clone git@github.com:Microsoft/vscode-java-test.git`
+2. `cd vscode-java-test/extension`
+3. Install the dependencies: `npm install`
+4. Build the Java Language Server plugin for VS Code Java Test Runner: `gulp build_server`
+5. Open the folder in VS Code
+> Note: The below steps are only required if you want to debug the Java codes of the extension, if you are only interested in the TypeScript part, you are done now
+6. Download the [newest bits of the Java Language Server](http://download.eclipse.org/jdtls/snapshots/jdt-language-server-latest.tar.gz)
+7. Decompress the downloaded bits, you will get a folder named `jdt-language-server-latest`
```
Comment (yaohaizh) [inline CONTRIBUTING.md]: Should add a target.file directly

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+8. Copy the folder `jdt-language-server-latest` into the project base path
+9. Import project in Eclipse
```
Comment (ansyral) [inline CONTRIBUTING.md]: I'm thinking how about use vscode?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
+## First Time Setup
+1. Fork and clone the repository: `git clone git@github.com:Microsoft/vscode-java-test.git`
+2. `cd vscode-java-test/extension`
+3. Install the dependencies: `npm install`
+4. Build the Java Language Server plugin for VS Code Java Test Runner: `gulp build_server`
```
Comment (ansyral) [inline CONTRIBUTING.md]: and the runner jars?

File: CONTRIBUTING.md (hunk 2 above)
Comment (jdneo) [reply CONTRIBUTING.md]: Tracking by #282

File: CONTRIBUTING.md (hunk 3 above)
Comment (jdneo) [reply CONTRIBUTING.md]: Tracked by #284

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
+## Setup
+1. Fork and clone the repository: `git clone git@github.com:Microsoft/vscode-java-test.git`
+2. `cd vscode-java-test/extension`
+3. Install the dependencies: `npm install`
+4. Build the Java Test Runner bundle for Java Language Server and the runner jars: `gulp build-jars`
```
Comment (akaroml) [inline CONTRIBUTING.md]: To run gulp in cli, you'll need to `install gulp-cli -g`.
So I would suggest that we use [npm scripts](https://docs.npmjs.com/misc/scripts) as the interface. You can refer to how a blank vscode extension project.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
+## Setup
```
Comment (akaroml) [inline CONTRIBUTING.md]: The current content covers the process of setup dev environment. But we also need to describe the dev process, contribution types, and channels. Please take [VS Code Contribution Guide](https://github.com/Microsoft/vscode/blob/master/CONTRIBUTING.md) as an example.

Comment (akaroml) [reply CONTRIBUTING.md]: Another area we need to cover is the design. This extension consists of TypeScript code and Java code. We need to explain what each part does and how they work.

Comment (akaroml) [reply CONTRIBUTING.md]: I would suggest that we agree on a table of contents first. Then you can fill in contents for each topic.

**Comment by jdneo** [comment]
@akaroml Thanks for the comments.

I'll revisit this PR later. Just focusing on some issues which have higher priority.

**Pull request closed without merging**